### ISSUE TITLE
Handle offline state when adding accounts

### ISSUE THREAD

**Issue by Mehdi-git** [stefan-niedermann/nextcloud-notes#1014]
When a user is offline and tries to add an account, the app crashes (`UnknownErrorException: Unable to resolve host "cloud.example.com": No address associated with hostname`).

The app should validate that the device is connected to the internet before proceeding and prompting an proper error message in case the device is offline.

<details>
<summary>Original post</summary>
Hello,
When note app try to add account from next-cloud, after select account and ask for permition to access account, app get crashed when there is no internet connectivity. 
In next-cloud deck app, user can not add account when there is no internet connectivity so this problem never happened. 

```
App Version: 2.18.1
App Version Code: 2018001
App Flavor: dev

Files App Version Code: 30150000

---

OS Version: 4.9.118+(59)
OS API Level: 29
Device: Aphone_6s
Manufacturer: Apex
Model (and Product): Aphone_6s (Aphone_6s)

---

com.nextcloud.android.sso.exceptions.UnknownErrorException: Unable to resolve host "cloud.batna.ir": No address associated with hostname
	at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:172)
	at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:171)
	at it.niedermann.owncloud.notes.persistence.SSOClient.requestFilesApp(SSOClient.java:27)
	at it.niedermann.owncloud.notes.persistence.CapabilitiesClient.getCapabilities(CapabilitiesClient.java:64)
	at it.niedermann.owncloud.notes.main.MainActivity.lambda$null$20$MainActivity(MainActivity.java:791)
	at it.niedermann.owncloud.notes.main.-$$Lambda$MainActivity$NkqJ_xHD_VohPbcetf0ICKMv8pA.run(Unknown Source:4)
	at java.lang.Thread.run(Thread.java:919)

```
Thank you.
</details>

**Comment by Cui-Yusong** [comment]
May I ask if this problem can be solved in this way
When adding an account, first let the application determine whether it is connected to the network, if the connection is successful, then can correctly add. If there is no network connection, the program will not add account and prompt the user to connect to the network, so as to avoid the program crash.
Is this idea feasible?

**Comment by stefan-niedermann** [comment]
This was my first thought, yes. I eventually [went the shorter way](https://github.com/stefan-niedermann/nextcloud-notes/commit/71010248cdd0941d9a68ab94874d96cc23941d62) to catch the exception and just display a human readable hint when it is network related:

![grafik](https://user-images.githubusercontent.com/4741199/115993062-450a8f80-a5d1-11eb-939f-e44bab35b205.png)

This is also how we are doing it in the Deck app, so that's fine for the **`ImportAccountActivity`**.

To finally fix this issue, we need also take care of this case in when adding a second or nth account from within the **`MainActivity`**. You can add a further `if / else` block for our `UnknownErrorException` [here](https://github.com/stefan-niedermann/nextcloud-notes/blob/master/app/src/main/java/it/niedermann/owncloud/notes/main/MainActivity.java#L645-L658) and handle it with a meaningful snackbar like `R.string.error_no_network` (this string is already available) instead of prompting the `ExceptionDialogFragment` :slightly_smiling_face:

**Comment by Cui-Yusong** [comment]
Can I read it this way:
When a user wants to add an account and this causes it to go to Unknowner RoleException because it is not connected, let application determine whether it is connected to the Internet. If there is no connection to the network, the user will be prompted like R.string.error_no_network to hint the user to connect to the network.

**Comment by stefan-niedermann** [comment]
Basically yes. No need to determine the connection - if you catch the `UnknownErrorException` then it already happend and we already know that something went wrong with the connection.

So just catch it and then show a snackbar 🙂

**Comment by stefan-niedermann** [comment]
Fixed in `v3.4.1`
